packages/twenty-front/src/modules/object-record/multiple-objects/hooks/usePerformCombinedFindManyRecords.ts

```diff
@@ -0,0 +1,225 @@
+import { ApolloClient, gql, useApolloClient } from '@apollo/client';
+import { isUndefined } from '@sniptt/guards';
+import { capitalize, isDefined } from 'twenty-shared';
+import { isNonEmptyArray } from '~/utils/isNonEmptyArray';
+
+import { objectMetadataItemsState } from '@/object-metadata/states/objectMetadataItemsState';
+import { ObjectMetadataItem } from '@/object-metadata/types/ObjectMetadataItem';
+import { mapObjectMetadataToGraphQLQuery } from '@/object-metadata/utils/mapObjectMetadataToGraphQLQuery';
+import { getRecordsFromRecordConnection } from '@/object-record/cache/utils/getRecordsFromRecordConnection';
+import { EMPTY_QUERY } from '@/object-record/constants/EmptyQuery';
+import { RecordGqlOperationSignature } from '@/object-record/graphql/types/RecordGqlOperationSignature';
+import { generateDepthOneRecordGqlFields } from '@/object-record/graphql/utils/generateDepthOneRecordGqlFields';
+import { CombinedFindManyRecordsQueryResult } from '@/object-record/multiple-objects/types/CombinedFindManyRecordsQueryResult';
+import { getCombinedFindManyRecordsQueryFilteringPart } from '@/object-record/multiple-objects/utils/getCombinedFindManyRecordsQueryFilteringPart';
+import { useRecoilValue } from 'recoil';
+
+export const usePerformCombinedFindManyRecords = () => {
+  const client = useApolloClient();
+  const objectMetadataItems = useRecoilValue(objectMetadataItemsState);
+
+  const generateCombinedFindManyRecordsQuery = (
```

Comment:
Can we re-use `useGenerateCombinedFindManyRecordsQuery` for this ? It would be interesting to put `generateCombinedFindManyRecordsQuery` into the hook call ?